File: src/main/scala/mrtjp/fengine/simulate/ICSimulationCallback.scala
package mrtjp.fengine.simulate

trait ICSimulationCallback
{
    def registersDidChange(registers:Set[Int]):Unit

    def icDidThrowErrorFlag(flag:Int, registers:Seq[Int], gates:Seq[Int]):Unit

    def icEventComputeOverflow(registers:Seq[Int], gates:Seq[Int], computeLimit:Int):Unit
}

File: src/main/java/mrtjp/fengine/simulate/ICSimulationCallback.java
package mrtjp.fengine.simulate;

import java.util.Set;

public interface ICSimulationCallback {

    void registersDidChange(Set<Integer> registers);

    void icDidThrowErrorFlag(int flag, Set<Integer> registers, Set<Integer> gates);

    void icEventComputeOverflow(Set<Integer> registers, Set<Integer> gates, int computeLimit);
}
